Source/WebCore/rendering/style/StyleRareInheritedData.h

```diff
@@ -204,6 +205,9 @@ class StyleRareInheritedData : public RefCounted<StyleRareInheritedData> {
 
     WordBoundaryDetection wordBoundaryDetection;
 
+    // Investigate using Markable here to save on memory
```

Comment:
This would be very easy to do since `StyleColor` has a null value, and I don’t see why we wouldn’t do it right away instead of just leaving a comment.

Reply:
I'm not sure how to go about doing this? When I tried adding a MarkableTraits struct to the ScrollbarColor struct I got errors such as `Incomplete type 'WebCore::ScrollbarColor' named in nested name specifier` I believe this is something to do with how the ScrollbarColor type is forwarded inside of RenderStyle.h. I had other related errors too

(I'm aware the below code is almost definitely not fully correct)

```cpp
struct MarkableTraits {
        static bool isEmptyValue(const ScrollbarColor& value)
        {
            return value.thumbColor == StyleColor { Color(std::nullopt) } && value.trackColor == StyleColor(Color(std::nullopt));
        }

        static ScrollbarColor emptyValue()
        {
            return ScrollbarColor {
                StyleColor { Color(std::nullopt) },
                StyleColor { Color(std::nullopt) }
            };
        }
    };
```

Reply:
I'll push up the other requested changes such as changing to a struct and the code cleanup. And then hopefully someone can take a look at this to point me in the right direction.

Reply:
There are really two separate problems:
1) How to declare the markable traits so they will be used correctly in `RenderStyle.h`. One solution there is to have the functions return `std::optional<ScrollbarColor>` instead of `const Markable< ScrollbarColor>&`, which probably will work fine from a performance point of view and doesn’t require any markable traits in that header at all. It’s worth looking for another example of use of `Markable` in `RenderStyle` for a view on what idiom they use.
2) How to define the markable traits in the `ScrollbarColor.h` header so they work correctly for **use** of `Markable<StyleColor>`.

It should be closer to this (but this is not exactly right):
```
    template<> struct MarkableTraits<StyleColor> {
        static bool isEmptyValue(const ScrollbarColors& value) { !return value.thumbColor.isValid(); };
        static ScrollbarColors emptyValue() { return { } };
    }
```
Might need to add some `{ }` to the definition of the ScrollbarColors struct to make this work.

Reply:
I couldn't find an example of Markable in RenderStyle but I *think* I've worked it out. The tests seemed to pass locally so will see what the pipeline thinks of it.